pkg/controller/acmeorders/sync_test.go

```diff
@@ -229,90 +223,100 @@ dGVzdA==
+			builder: &testpkg.Builder{
+				CertManagerObjects: []runtime.Object{testIssuerHTTP01Enabled, testOrderPending, testAuthorizationChallengeInvalid},
 				ExpectedActions:    []testpkg.Action{},
 			},
-			Client: &acmecl.FakeACME{
+			acmeClient: &acmecl.FakeACME{
 				FakeGetOrder: func(_ context.Context, url string) (*acmeapi.Order, error) {
 					return testACMEOrderPending, nil
 				},
 			},
-			CheckFn: func(t *testing.T, s *controllerFixture, args ...interface{}) {
-			},
-			Err: false,
 		},
 		"do nothing if the order is valid": {
-			Issuer: testIssuerHTTP01Enabled,
-			Order:  testOrderValid,
-			Builder: &testpkg.Builder{
-				CertManagerObjects: []runtime.Object{testOrderValid},
+			order: testOrderValid,
+			builder: &testpkg.Builder{
+				CertManagerObjects: []runtime.Object{testIssuerHTTP01Enabled, testOrderValid},
 				ExpectedActions:    []testpkg.Action{},
 			},
-			Client: &acmecl.FakeACME{},
-			CheckFn: func(t *testing.T, s *controllerFixture, args ...interface{}) {
-			},
-			Err: false,
+			acmeClient: &acmecl.FakeACME{},
 		},
 		"do nothing if the order is failed": {
-			Issuer: testIssuerHTTP01Enabled,
-			Order:  testOrderInvalid,
-			Builder: &testpkg.Builder{
-				CertManagerObjects: []runtime.Object{testOrderInvalid},
+			order: testOrderInvalid,
+			builder: &testpkg.Builder{
+				CertManagerObjects: []runtime.Object{testIssuerHTTP01Enabled, testOrderInvalid},
 				ExpectedActions:    []testpkg.Action{},
 			},
-			Client: &acmecl.FakeACME{},
-			CheckFn: func(t *testing.T, s *controllerFixture, args ...interface{}) {
-			},
-			Err: false,
+			acmeClient: &acmecl.FakeACME{},
 		},
 	}
 
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
-			if test.Builder == nil {
-				test.Builder = &testpkg.Builder{}
-			}
-			if test.Clock == nil {
-				test.Clock = fixedClock
-			}
-			test.Setup(t)
-			orderCopy := test.Order.DeepCopy()
-			err := test.Controller.Sync(test.Ctx, orderCopy)
-			if err != nil && !test.Err {
-				t.Errorf("Expected function to not error, but got: %v", err)
+			// reset the fixedClock at the start of each test
+			fixedClock.SetTime(nowTime)
+			// always use the fixedClock unless otherwise specified
+			if test.builder.Clock == nil {
+				test.builder.Clock = fixedClock
 			}
-			if err == nil && test.Err {
-				t.Errorf("Expected function to get an error, but got: %v", err)
-			}
-			test.Finish(t, orderCopy, err)
+			runTest(t, test)
 		})
 	}
 }
 
+type testT struct {
+	order      *v1alpha1.Order
+	builder    *testpkg.Builder
+	acmeClient acmecl.Interface
+	expectErr  bool
+}
+
+func runTest(t *testing.T, test testT) {
+	test.builder.T = t
+	test.builder.Start()
+	defer test.builder.Stop()
+
+	c := &controller{}
+	c.Register(test.builder.Context)
+	c.helper = issuer.NewHelper(
+		test.builder.SharedInformerFactory.Certmanager().V1alpha1().Issuers().Lister(),
+		test.builder.SharedInformerFactory.Certmanager().V1alpha1().ClusterIssuers().Lister(),
+	)
```

Comment:
I've updated this test to no longer use a 'fake' helper here, which actually masked a few misconfigurations in the test fixtures (i.e. the issuer resource not actually existing in the apiserver).